src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/DelegationHelper.cs

```diff
@@ -275,5 +275,45 @@ public static string GetAttributeMatchKey(List<AttributeMatch> attributeMatches)
         {
             return string.Concat(attributeMatches.OrderBy(r => r.Id).Select(r => r.Id + r.Value));
         }
+
+        /// <summary>
+        /// Sets the RuleType on each rule in the given list
+        /// </summary>
+        public static void SetRuleType(List<Rule> rulesList, List<int> keyRolePartyIds, List<AttributeMatch> coveredBy, int parentPartyId = 0)
+        {
+            foreach (Rule rule in rulesList)
+            {
+                if (TryGetDelegationParamsFromRule(rule, out _, out _, out _, out int? coveredByPartyId, out int? coveredByUserId, out _)
+                    && rule.Type == RuleType.None)
+                {
+                    SetTypeForSingleRule(keyRolePartyIds, coveredBy, parentPartyId, rule, coveredByPartyId, coveredByUserId);
+                }
+            }
+        }
+
+        private static void SetTypeForSingleRule(List<int> keyRolePartyIds, List<AttributeMatch> coveredBy, int parentPartyId, Rule rule, int? coveredByPartyId, int? coveredByUserId)
+        {
+            if ((TryGetCoveredByUserIdFromMatch(coveredBy, out int coveredByUserIdFromRequest) && coveredByUserIdFromRequest == coveredByUserId)
```

Comment:
It's not enough to only check the CoveredBy the OfferedBy must also be identical to the request for it to be a DirectDelegation.
Otherwise InheritedAsSubunit rules (where the OfferedBy is the PartyParentId) will be falsely matched here.

Reply:
Fixed. Offeredby is also checked now.